counsel.el

```diff
@@ -1919,25 +1919,38 @@ INITIAL-INPUT can be given as the initial minibuffer input."
   nil)
 
 ;;;###autoload
-(defun counsel-fzf (&optional initial-input)
-  "Call the \"fzf\" shell command.
-INITIAL-INPUT can be given as the initial minibuffer input."
-  (interactive)
-  (counsel-require-program (car (split-string counsel-fzf-cmd)))
-  (setq counsel--fzf-dir
-        (if (and
-             (fboundp 'projectile-project-p)
-             (fboundp 'projectile-project-root)
-             (projectile-project-p))
-            (projectile-project-root)
-          default-directory))
-  (ivy-read "> " #'counsel-fzf-function
-            :initial-input initial-input
-            :re-builder #'ivy--regex-fuzzy
-            :dynamic-collection t
-            :action #'counsel-fzf-action
-            :unwind #'counsel-delete-process
-            :caller 'counsel-fzf))
+(defun counsel-fzf (&optional initial-input initial-directory fzf-prompt)
+  "Open a file using the fzf shell command.
+INITIAL-INPUT can be given as the initial minibuffer input.
+INITIAL-DIRECTORY, if non-nil, is used as the root directory for search.
+FZF-PROMPT, if non-nil, is passed as `ivy-read' prompt argument."
+  (interactive
+   (let ((fzf-basename (car (split-string counsel-fzf-cmd))))
+     (list nil
+           (when current-prefix-arg
+             (read-directory-name (concat
+                                   fzf-basename
+                                   " in directory: "))))))
+
+  (let ((fzf-basename (car (split-string counsel-fzf-cmd))))
+    (counsel-require-program fzf-basename)
+    (setq counsel--fzf-dir
+          (if (and
+               (fboundp 'projectile-project-p)
+               (fboundp 'projectile-project-root)
+               (projectile-project-p))
+              (projectile-project-root)
+            (or initial-directory
+                default-directory)))
+    (ivy-set-prompt 'counsel-ag counsel-prompt-function)
```

Comment:
Should this be counsel-fzf?

Reply:
yes it should, not sure how that slipped in, thanks for the review, let me fix it